zenoh-plugin-ros2dds/src/route_service_srv.rs

```diff
@@ -461,32 +462,9 @@ fn route_dds_reply_to_zenoh(
     // Check if it's one of my queries in progress. Drop otherwise
     match queries_in_progress.remove(&request_id) {
         Some(query) => {
-            // copy CDR Header
-            let cdr_header = match dds_rep_buf.get(0..4) {
-                Some(hdr) => hdr,
-                None => {
-                    // TODO: do we want this to return or explode ?
-                    tracing::error!(
-                        "Sample buffer was empty, Could not ger cdr header expected bytes [0..4]",
-                    );
-                    return;
-                }
-            };
-
-            // copy Request payload, skipping client_id + sequence_number
-            let payload = match dds_rep_buf.get(20..dds_rep_buf.len()) {
-                Some(hdr) => hdr,
-                None => {
-                    // TODO: do we want this to return or explode ?
-                    tracing::error!(
-                        "Sample buffer payload subslice failed, expected bytes [20..buffer.len()]",
-                    );
-                    return;
-                }
-            };
-
-            let res: Vec<u8> = [cdr_header, payload].concat();
-            let zenoh_rep_buf = ZBytes::new(res);
+            let mut zenoh_rep_buf = ZBuf::empty();
+            zenoh_rep_buf.push_zslice(slice.subslice(0, 4).unwrap());
+            zenoh_rep_buf.push_zslice(slice.subslice(20, slice.len()).unwrap());
```

Comment:
The same could be done here as above, explicitly handling the subslice `Option` variants of ZSlice.
Again, more verbose, but I leave it up to you.

Reply:
Addressed in latest commit. Please review!